node/core/pvf/common/src/worker/mod.rs

```diff
@@ -317,4 +317,92 @@ pub mod thread {
 			Some(*result.0)
 		}
 	}
+
+	#[cfg(test)]
+	mod tests {
+		use super::*;
+		use crate::worker::cpu_time_monitor_loop;
+		use cpu_time::ProcessTime;
+		use std::sync::mpsc::channel;
+
+		#[test]
+		fn get_condvar_should_be_pending() {
+			let condvar = get_condvar();
+			let outcome = *condvar.0.lock().unwrap();
+			assert!(outcome.is_pending());
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_return_none_on_time_out() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(1));
+			assert!(matches!(outcome, None));
+		}
+
+		#[test]
+		fn wait_for_threads_with_timeout_returns_outcome() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let condvar2 = Arc::clone(&condvar);
+			thread::spawn(move || {
+				cond_notify_all(Arc::clone(&condvar2), WaitOutcome::Finished);
+			});
+			let outcome = wait_for_threads_with_timeout(&condvar, Duration::from_secs(2));
+			assert!(matches!(outcome.unwrap(), WaitOutcome::Finished));
+		}
+
+		#[test]
+		fn spawn_worker_thread_should_notify_on_done() {
+			let condvar = Arc::new((Mutex::new(WaitOutcome::Pending), Condvar::new()));
+			let response = spawn_worker_thread(
+				"thread",
+				move || 2,
```

Comment:
Nit: You don't have to `move` here (as there's nothing to move besides the primitive value, which is `Copy`).